mlrun/feature_store/retrieval/job.py

```diff
@@ -72,9 +73,13 @@ def run_merge_job(
             "drop_columns": drop_columns,
             "with_indexes": with_indexes,
             "query": query,
+            "join_type": join_type,
             "engine_args": engine_args,
         },
         inputs={"entity_rows": entity_rows},
+        out_path=function.spec.output_path
+        if hasattr(function.spec, "output_path")
+        else None,
```

Comment:
This was added as part of a git merge at some point, I assume? Well, it's not needed and needs to go - you can see that @gtopper reverted this in #2902 - so it can be removed (no real function has `output_path` in its `spec`)